.github/workflows/deploy-plugins.yml

```diff
@@ -101,6 +98,10 @@ jobs:
 
           echo "deploy=true" >> $GITHUB_OUTPUT
 
+      - name: Check plugin version integrity
+        if: steps.check-deployment.outputs.deploy == 'true'
```

Comment:
Not blocker: Instead of adding `if: steps.check-deployment.outputs.deploy == 'true'` condition in every steps can we add new step if output is not true the exit and remove condition from all other steps below?

Reply:
Not sure how that would work in GitHub Actions. Maybe it's possible, but I'm not sure there's a way to properly exit out of a GitHub workflow. Do you have an idea about it?

Reply:
@swissspidy Curious what's your take on the idea of early returns/exits in GitHub workflows.

Reply:
Ping @swissspidy :)

Reply:
Thanks for the ping. I was on vacation when you sent the first message.

It depends on what you want to achieve.

If you `exit 1` early that will mark the whole job as failed. So the question is whether that is desirable.

Also depends on if there are any other steps after the exit 1 that still should be executed. Then obviously you can't exit early.

IMO the behavior at the moment is quite nice. The jobs are marked as passed, but you still get the notice saying no deployment was needed:

<img width="1213" alt="Screenshot 2024-12-18 at 09 57 41" src="https://github.com/user-attachments/assets/aee2f61e-ba9e-4850-b367-9132090e835a" />

Whereas with the suggested early exit you'll get a failure status and my first instinct is "oh oh, something is broken" and not "oh, this is just a skipped plugin"

Reply:
That's a great point, and I agree. We shouldn't exit with a failure code, as skipping the deployment because it's not needed is indeed not a failure.